src/ImageSharp/Formats/Ani/AniDecoderCore.cs

```diff
@@ -0,0 +1,416 @@
+// Copyright (c) Six Labors.
+// Licensed under the Six Labors Split License.
+
+using System.Buffers;
+using System.Collections;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+using SixLabors.ImageSharp.Formats.Bmp;
+using SixLabors.ImageSharp.Formats.Cur;
+using SixLabors.ImageSharp.Formats.Ico;
+using SixLabors.ImageSharp.Formats.Icon;
+using SixLabors.ImageSharp.Formats.Webp;
+using SixLabors.ImageSharp.IO;
+using SixLabors.ImageSharp.Memory;
+using SixLabors.ImageSharp.Memory.Internals;
+using SixLabors.ImageSharp.Metadata;
+
+namespace SixLabors.ImageSharp.Formats.Ani;
+
+internal class AniDecoderCore(DecoderOptions options) : ImageDecoderCore(options)
+{
+    /// <summary>
+    /// The general decoder options.
+    /// </summary>
+    private readonly Configuration configuration = options.Configuration;
+
+    /// <summary>
+    /// The stream to decode from.
+    /// </summary>
+    private BufferedReadStream currentStream = null!;
+
+    private AniHeader header;
+
+    private enum ListIconChunkType : byte
+    {
+        Ico = 1,
+        Cur,
+        Bmp
+    }
+
+    protected override Image<TPixel> Decode<TPixel>(BufferedReadStream stream, CancellationToken cancellationToken)
+    {
+        this.currentStream = stream;
+
+        Guard.IsTrue(this.currentStream.TryReadUnmanaged(out RiffOrListChunkHeader riffHeader), nameof(riffHeader), "Invalid RIFF header.");
+        long dataSize = riffHeader.Size;
+        long dataStartPosition = this.currentStream.Position;
+
+        ImageMetadata metadata = new();
+        AniMetadata aniMetadata = this.ReadHeader(dataStartPosition, dataSize, metadata);
+
+        List<(ListIconChunkType Type, Image<TPixel> Image)> frames = [];
+        this.HandleRiffChunk(out Span<int> sequence, out Span<uint> rate, dataStartPosition, dataSize, aniMetadata, frames, DecodeFrameChunk);
+
+        List<ImageFrame<TPixel>> list = [];
+
+        foreach (int i in sequence)
```

Comment:
We need to flatten out the frames and metadata here. Each frame should have an `AniFrameMetadata` property containing all the required information including what sequence the frame belongs to. 

No other format metadata should be added to the output.

Reply:
How can i remove the FrameMetadata from decoded frame? Can I know if a Metadata is included without creating a new Metadata?